src/stateless/MiddleEllipsis.vue

```diff
@@ -0,0 +1,72 @@
+<template>
+    <div ref="middle-ellipsis" class="middle-ellipsis">
+        <div ref="measurement-width" class="middle-ellipsis__measurement">{{ text }}</div>
+        <div :title="text" class="middle-ellipsis__text">{{ truncatedText }}</div>
+    </div>
+</template>
+
+<script>
+function middleEllipsis (value, breakAt, middle = breakAt / 2) {
+    if (value.length > breakAt) {
+        const ellipsisLength = 4
+        return (
+            value.substring(0, breakAt - middle - ellipsisLength) +
+            '…' +
+            value.substring(value.length - middle, value.length)
+        )
+    }
+    return value
+}
+
+export default {
+    props: {
+        text: {
+            required: true,
+            type: String,
+        },
+    },
+    data () {
+        return {
+            actualWidth: 0,
+        }
+    },
+    computed: {
+        truncatedText () {
+            const length = this.text.length
+            if (this.actualWidth > 0 && length > 4) {
+                const percent = this.width / this.actualWidth
+                const breakAt = Math.floor(length * percent)
+
+                if (percent < 1) {
+                    return middleEllipsis(this.text, breakAt)
+                }
+            }
+
+            return this.text
+        },
+    },
+    mounted () {
+        this.width = parseInt(window.getComputedStyle(this.$refs['middle-ellipsis']).width, 10)
```

Comment:
Maybe you should also update this on resize event and make it reactive.

Reply:
Maybe, but I'd only add it when it's needed, not before.